tasks/install_shell.sh

```diff
@@ -398,6 +411,20 @@ install_file() {
       ;;
     "noarch.rpm")
       info "installing puppetlabs yum repo with zypper..."
+
+      if test "x$installed_version" != "xuninstalled"; then
```

Comment:
Currently if the puppet5-release repo is installed dpkg, zypper, or rpm is used to remove it. This allows upgrade to puppet6. However going the "other way" seems like it would also require an apt-get, yum or zypper to also uninstall puppet-agent. Should we include this logic?

Reply:
I think downgrades through collections can error.